templates/ingress/ingressroute.yaml

```diff
@@ -0,0 +1,43 @@
+{{- if eq .Values.expose.type "ingressRoute" }}
+{{- $ingressRoute := .Values.expose.ingressRoute -}}
+{{- $_ := set . "portal_path" "/" -}}
+{{- $_ := set . "api_path" "/api/" -}}
+{{- $_ := set . "service_path" "/service/" -}}
+{{- $_ := set . "v2_path" "/v2/" -}}
+{{- $_ := set . "chartrepo_path" "/chartrepo/" -}}
+{{- $_ := set . "controller_path" "/c/" -}}
+apiVersion: traefik.containo.us/v1alpha1
+kind: IngressRoute
+metadata:
+  name: "{{ template "harbor.ingressRoute" . }}"
+  labels:
+{{ include "harbor.labels" . | indent 4 }}
+  annotations:
+{{ toYaml $ingressRoute.annotations | indent 4 }}
+spec:
+  entryPoints:
+{{ toYaml $ingressRoute.entryPoints | indent 2 }}
+  routes:
+  - kind: Rule
+    match: Host(`{{ $ingressRoute.hosts.core }}`)
+    services:
+    - kind: Service
+      name: {{ template "harbor.portal" . }}
+      port: 80
+  - kind: Rule
+    match: Host(`{{ $ingressRoute.hosts.core }}`) && (PathPrefix(`{{ .api_path }}`) || PathPrefix(`{{ .service_path }}`) || PathPrefix(`{{ .v2_path }}`) || PathPrefix(`{{ .chartrepo_path }}`) || PathPrefix(`{{ .controller_path }}`))
+    services:
+    - kind: Service
+      name: {{ template "harbor.core" . }}
+      port: 80
+  {{- if .Values.notary.enabled }}
+  - kind: Rule
+    match: Host(`{{ $ingressRoute.hosts.notary }}`)
+    services:
+    - kind: Service
+      name: {{ template "harbor.notary-server" . }}
+      port: 4443
+  {{- end }}
+  tls: 
```

Comment:
Might want to wrap `tls:` in a conditional so the rendered file does not have it listed when tls is empty. Traefik still reads the empty object and thinks tls is enabled on the router which does not work if you are terminating at an external load balancer.

Reply:
thanks for that correction 👍